File: src/ChargePadLine.Entitys/Trace/BOM/BomItem.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using ChargePadLine.Entitys.Systems;

namespace ChargePadLine.Entitys.Trace.BOM
{
  /// <summary>
  /// 工单
  /// </summary>|
  [Table("mes_bom_item")]
  public class BomItem : BaseEntity
  {
    /// <summary>
    /// BOM明细ID
    /// </summary>
    [Key]
    [Column("BomItemId")]
    public Guid BomItemId { get; set; }

    /// <summary>
    /// 主表ID
    /// </summary>
    [Column("BomId")]
    public Guid? BomId { get; set; }

    /// <summary>
    /// 站点编码
    /// </summary>
    [Column("StationCode")]
    public string StationCode { get; set; }

    /// <summary>
    /// 批次规则，正则表达
    /// </summary>
    [Column("BatchRule")]
    public string BatchRule { get; set; }

    /// <summary>
    /// 批次数据固定
    /// </summary>
    [Column("BatchQty")]
    public bool? BatchQty { get; set; }

    /// <summary>
    /// 获取批次数量，固定位数
    /// </summary>
    [Column("BatchSNQty")]
    public string BatchSNQty { get; set; }

    /// <summary>
    /// 物料ID
    /// </summary>
    [Column("ProductId")]
    public Guid? ProductId { get; set; }

    /// <summary>
    /// 日期序列
    /// </summary>
    [Column("DateIndex")]
    public int? DateIndex { get; set; }

    /// <summary>
    /// 数量序列
    /// </summary>
    [Column("NumberIndex")]
    public int? NumberIndex { get; set; }

    /// <summary>
    /// 物料保质期（分钟）
    /// </summary>
    [Column("ShelfLife")]
    public int? ShelfLife { get; set; }

    /// <summary>
    /// 物料号序列
    /// </summary>
    [Column("ProductIndex")]
    public int? ProductIndex { get; set; }

    /// <summary>
    /// BOM主表（多对一关联）
    /// </summary>
    [ForeignKey("BomId")]
    public virtual BomList Bom { get; set; }
  }

}

File: web_client_pro/src/services/Model/Infrastructure/Bom/BomItem.ts
/**
 * BOM子项相关类型定义
 */

/**
 * BOM子项查询DTO
 */
export interface BomItemQueryDto {
  /**
   * 当前页码
   */
  current: number;
  /**
   * 每页大小
   */
  pageSize: number;
  /**
   * 排序字段
   */
  sortField?: string;
  /**
   * 排序顺序
   */
  sortOrder?: string;
  /**
   * BOM ID
   */
  bomId?: string;
  /**
   * 站点编码
   */
  stationCode?: string;
  /**
   * 批次规则
   */
  batchRule?: string;
  /**
   * 批次SN数量
   */
  batchSNQty?: string;
  /**
   * 物料ID
   */
  productId?: string;

  /**
   * 产品名称
   */
  productName?: string;

  /**
   * 产品编码
   */
  productCode?: string;
/**
 * 日期序列
 */
dataIndex?: number;
/**
 * 数量序列
 */
numberIndex?: number;
/**
 * 物料保质期（分钟）
 */
shelfLife?: number;
/**
 * 物料号序列
 */
dateIndex?: number;
}

/**
 * BOM子项DTO
 */
export interface BomItemDto {
  /**
   * BOM明细ID
   */
  bomItemId: string;

  /**
   * 主表ID
   */
  bomId?: string;

  /**
   * 站点编码
   */
  stationCode: string;

  /**
   * 批次规则，正则表达
   */
  batchRule: string;

  /**
   * 批次数据固定
   */
  batchQty?: boolean;

  /**
   * 获取批次数量，固定位数
   */
  batchSNQty: string;

  /**
   * 物料ID
   */
  productId?: string;

  /**
   * 产品名称
   */
  productName?: string;

  /**
   * 产品编码
   */
  productCode?: string;
  /**
 * 日期序列
 */
dataIndex?: number;
/**
 * 数量序列
 */
numberIndex?: number;
/**
 * 物料保质期（分钟）
 */
shelfLife?: number;
/**
 * 物料号序列
 */
dateIndex?: number;

  /**
   * BOM主表（多对一关联）
   */
  bom?: import('./BomList').BomList;

  /**
   * 搜索值
   */
  searchValue?: string;

  /**
   * 创建者
   */
  createBy?: string;

  /**
   * 创建时间
   */
  createTime?: string;

  /**
   * 更新者
   */
  updateBy?: string;

  /**
   * 更新时间
   */
  updateTime?: string;

  /**
   * 请求参数
   */
  params?: Record<string, any>;
}

/**
 * BOM子项创建DTO
 */
export interface BomItemCreateDto {
  /**
   * 主表ID
   */
  bomId?: string;

  /**
   * 站点编码
   */
  stationCode: string;

  /**
   * 批次规则，正则表达
   */
  batchRule: string;

  /**
   * 批次数据固定
   */
  batchQty?: boolean;

  /**
   * 获取批次数量，固定位数
   */
  batchSNQty: string;

  /**
   * 物料ID
   */
  productId?: string;

  /**
   * 产品名称
   */
  productName?: string;

  /**
   * 产品编码
   */
  productCode?: string;
  /**
 * 产品序列
 */
productIndex?: number;
/**
 * 数量序列
 */
numberIndex?: number;
/**
 * 物料保质期（分钟）
 */
shelfLife?: number;
/**
 * 物料号序列
 */
dateIndex?: number;
}

/**
 * BOM子项更新DTO
 */
export interface BomItemUpdateDto {
  /**
   * BOM明细ID
   */
  bomItemId: string;

  /**
   * 主表ID
   */
  bomId?: string;

  /**
   * 站点编码
   */
  stationCode: string;

  /**
   * 批次规则，正则表达
   */
  batchRule: string;

  /**
   * 批次数据固定
   */
  batchQty?: boolean;

  /**
   * 获取批次数量，固定位数
   */
  batchSNQty: string;

  /**
   * 物料ID
   */
  productId?: string;

  /**
   * 产品名称
   */
  productName?: string;

  /**
   * 产品编码
   */
  productCode?: string;
  /**
 * 产品序列
 */
productIndex?: number;
/**
 * 数量序列
 */
numberIndex?: number;
/**
 * 物料保质期（分钟）
 */
shelfLife?: number;
/**
 * 物料号序列
 */
dateIndex?: number;
}

/**
 * BOM子项实体（兼容API服务使用）
 */
export type BomItem = BomItemDto;
